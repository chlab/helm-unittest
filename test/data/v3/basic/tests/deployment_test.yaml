suite: test deployment
set:
  image:
    repository: apache
    tag: latest
    pullPolicy: Always
templates:
  - templates/configmap.yaml
  - templates/deployment.yaml
chart:
  version: 9.9.9+test
  appVersion: 9.9.9
tests:
  - it: should pass all kinds of assertion
    set:
      service.internalPort: 8080
    template: templates/deployment.yaml
    documentIndex: 0
    asserts:
      - equal:
          path: spec.template.spec.containers[?(@.name == "basic")].image
          value: apache:latest
      - notEqual:
          path: spec.template.spec.containers[?(@.name == "basic")].image
          value: nginx:stable
      - equal:
          path: metadata.labels.chart
          value: basic-9.9.9_test
      - equal:
          path: metadata.labels.appVersion
          value: 9.9.9
      - matchRegex:
          path: metadata.name
          pattern: ^.*-basic$
      - notMatchRegex:
          path: metadata.name
          pattern: ^.*-foobar$
      - equal:
          path: spec.template.spec.securityContext.runAsUser
          value: 1000
      - contains:
          path: spec.template.spec.containers[?(@.name == "basic")].ports
          content:
            containerPort: 8080
      - notContains:
          path: spec.template.spec.containers[?(@.name == "basic")].ports
          content:
            containerPort: 80
      - notExists:
          path: spec.template.nodeSelector
      - notExists:
          path: spec.template.spec.containers[1]
      - exists:
          path: spec.template
      - isNullOrEmpty:
          path: spec.template.spec.containers[?(@.name == "basic")].resources
      - isNotNullOrEmpty:
          path: spec.template.spec.containers[?(@.name == "basic")]
      - isKind:
          of: Deployment
      - isAPIVersion:
          of: extensions/v1beta1
      - hasDocuments:
          count: 2
      - hasDocuments:
          count: 1
          filterAware: true
      - matchSnapshot:
          path: spec
      - matchSnapshot:
          matchRegex:
            pattern: .*imagePullPolicy.*
          notMatchRegex:
            pattern: .*NetworkPolicy.*
      - isType:
          path: metadata.name
          type: string
      - lengthEqual:
          path: spec.template.spec.containers
          count: 1
      - notLengthEqual:
          path: spec.template.spec.containers
          count: 10

  - it: should pass greater or equal assertions
    set:
      runAsUser: 1284
      resources:
        requests:
          memory: "64Mi"
          cpu: 1.1
        limits:
          memory: "128Mi"
          cpu: "1500m"
    chart:
      appVersion: 0.1.0
    template: templates/deployment.yaml
    documentIndex: 0
    asserts:
      - greaterOrEqual:
          path: metadata.labels.appVersion
          value: 0.0.1
      - greaterOrEqual:
          path: spec.template.spec.securityContext.runAsUser
          value: 1000
      - greaterOrEqual:
          path: spec.template.spec.containers[?(@.name == "basic")].resources.limits.cpu
          value: "1440m"
      - greaterOrEqual:
          path: spec.template.spec.containers[?(@.name == "basic")].resources.requests.cpu
          value: 1.0
      - greaterOrEqual:
          path: spec.template.spec.containers[?(@.name == "basic")].resources.requests.cpu
          value: 1.1

  - it: should pass less or equal assertions
    set:
      runAsUser: 1284
      resources:
        requests:
          memory: "64Mi"
          cpu: 1.1
        limits:
          memory: "128Mi"
          cpu: "1500m"
    chart:
      appVersion: 10.0.0
    template: templates/deployment.yaml
    documentIndex: 0
    asserts:
      - lessOrEqual:
          path: metadata.labels.appVersion
          value: 10.0.0
      - lessOrEqual:
          path: spec.template.spec.securityContext.runAsUser
          value: 1285
      - lessOrEqual:
          path: spec.template.spec.containers[?(@.name == "basic")].resources.limits.cpu
          value: "2000m"
      - lessOrEqual:
          path: spec.template.spec.containers[?(@.name == "basic")].resources.requests.cpu
          value: 1.9
      - lessOrEqual:
          path: spec.template.spec.containers[?(@.name == "basic")].resources.requests.cpu
          value: 1.1
      - notLessOrEqual:
          path: spec.template.spec.containers[?(@.name == "basic")].resources.requests.cpu
          value: 1.0
      - notGreaterOrEqual:
          path: spec.template.spec.containers[?(@.name == "basic")].resources.requests.cpu
          value: 1.0
      - notGreaterOrEqual:
          path: spec.template.spec.containers[?(@.name == "basic")].resources.requests.cpu
          value: 1.2
      - notGreaterOrEqual:
          path: spec.template.spec.containers[?(@.name == "basic")].resources.requests.memory
          value: "1Mi"
      - notGreaterOrEqual:
          path: spec.template.spec.containers[?(@.name == "basic")].resources.requests.memory
          value: "101Mi"

  - it: should pass with no securityContext as runAsUser is removed
    set:
      runAsUser: null
    template: templates/deployment.yaml
    documentIndex: 0
    asserts:
      - notExists:
          path: spec.template.spec.securityContext
