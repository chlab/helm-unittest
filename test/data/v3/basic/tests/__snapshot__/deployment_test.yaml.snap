should pass all kinds of assertion:
  1: |
    replicas: 2
    template:
      metadata:
        labels:
          app: basic
          release: RELEASE-NAME
      spec:
        containers:
          - image: apache:latest
            imagePullPolicy: Always
            livenessProbe:
              httpGet:
                path: /
                port: 8080
            name: basic
            ports:
              - containerPort: 8080
            readinessProbe:
              httpGet:
                path: /
                port: 8080
            resources: {}
        securityContext:
          runAsUser: 1000
  2: |
    apiVersion: extensions/v1beta1
    kind: Deployment
    metadata:
      annotations:
        checksum/health: 5bde893f09313c69274774e2d93a68c7b5aa97256ea073a96832016c749b3f63
      labels:
        app: basic
        appVersion: 9.9.9
        chart: basic-9.9.9_test
        heritage: Helm
        release: RELEASE-NAME
      name: RELEASE-NAME-basic
    spec:
      replicas: 2
      template:
        metadata:
          labels:
            app: basic
            release: RELEASE-NAME
        spec:
          containers:
            - image: apache:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /
                  port: 8080
              name: basic
              ports:
                - containerPort: 8080
              readinessProbe:
                httpGet:
                  path: /
                  port: 8080
              resources: {}
          securityContext:
            runAsUser: 1000
